alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/transport/TransportIndexMonitorAction.kt

```diff
@@ -142,6 +149,41 @@ class TransportIndexMonitorAction @Inject constructor(
         })
     }
 
+    /**
+     * It's no reasonable to create AD monitor if the user has no access to any detector. Otherwise
+     * the monitor will not get any anomaly result. So we will check user has access to at least 1
+     * anomaly detector if they need to create AD monitor.
+     * As anomaly detector index is system index, common user has no permission to query. So we need
+     * to send REST API call to AD REST API.
+     */
+    fun checkAnomalyDetectorAndExecute(
+        client: Client,
+        actionListener: ActionListener<IndexMonitorResponse>,
+        request: IndexMonitorRequest
+    ) {
+        client.threadPool().threadContext.stashContext().use {
+            val searchSourceBuilder = SearchSourceBuilder().size(0)
```

Comment:
This will return 0 results since the size is 0

Reply:
Yeah, this is by design. We don't need to get the detail content of detector, just need the total hits. Check this line
```
val totalHits = response?.hits?.totalHits?.value
```

Reply:
ah ok